### ISSUE TITLE
query about using nuitka

### ISSUE THREAD

**Issue by innat-asj** [Nuitka/Nuitka#2388]
I came to know out Nuitka from this [post](https://varunbpatil.github.io/2019/09/29/nuitka-code-obfuscation.html) and wanted to use it to obfuscate some python scripts. In this post, same as cython or pyarmor, it is mentioned that Nuitka will generate `.so` file. But when I run it on a scripts (`main.py`), I don't see any generated `.so` file but getting two file, i.e. `main.bin` (file) and `main.build` (folder).  Could you please confirm what am I missing here?

**my env**

I am using Windows WSL (so basically Linux) to run Nuitka. 
Also I am using pyenv in my WSL for target python (3.6).

![qwe](https://github.com/Nuitka/Nuitka/assets/81627860/be70eaa5-bdcf-47b1-9ed6-9471110f18bd)

**Run**

This is how I run it.

![uio](https://github.com/Nuitka/Nuitka/assets/81627860/7d340492-01a9-4311-9dda-73e0d2ce2d47)

In the documentation, [here](https://nuitka.net/doc/user-manual.html#run-it), it is mentioned that a `.exe` file will be generated, I guess it is meant for windows 0S only.

**Additional query**

For obfuscation, primarily I picked cython. But the generated `.so` file compiled with **86x_64** architecture while I needed to run `.so` file on **arm64** device. So, unless I consider using cross-compilation on my 86x arch, I couldn't run `.so` on **arm64** devices. So, I was wondering if Nuitka is able to generate general compiled file that can be run on any architecture. Could you please confirm?

**Comment by innat-asj** [comment]
About `.so` file, from [this section](https://nuitka.net/doc/user-manual.html#use-case-2-extension-module-compilation), now I see it can be generated.

```python
python -m nuitka --module main.py
```

It will create 

```
main.cpython-36m-x86_64-linux-gnu.so'
```

Now, it looks like, same as cython, it is compiled with `x86_64` arch and will not run on `arm64` arch. Using Nuitka, will it be possible to make it general? As I don't have access `arm64` device, so unless I use cross-compilation, I couldn't generate `.so` type file from my local machine.

**Comment by kayhayen** [comment]
Thanks for sharing your reading results, I encourage you to read the docs and search the issue tracker as well. However, I cannot give individual help, unless you find problems in the docs, like the one pointing out, but when I checked it, I found you refer to the section "Tutorial Setup and build on Windows" which kind of points out, that sections for other OSes would be nice to have too, otherwise I find lots of places where the created binary suffix is pointed out to be OS and mode specifc.

For interactive questions, I prefer Discord, although there too, reading the docs is expected.
